### PR TITLE
more refactoring and updates to machine learning models

### PR DESCRIPTION
**Pull request by tomsilver**
* renames `torch_models` to `ml_models`
* add another layer of abstraction in the base classes for normalization
* add sklearn models
* update requirements.txt and mypy config for this new dependency
* should help with #685, but to close that issue, we will need to make changes in interactive learning approach too

### CODE REVIEW

File: src/ml_models.py [hunk 1]
```diff
@@ -726,17 +814,21 @@ def classify(self, x: Array) -> bool:
         assert classification in [True, False]
         return classification
 
-    def _fit(self, X: Array, y: Array) -> None:
```
Comment (ronuchit) [inline src/ml_models.py:729]: why is it okay to remove these?

File: src/ml_models.py [hunk 2]
```diff
@@ -8,11 +8,14 @@
...
 import tempfile
 from dataclasses import dataclass
-from typing import Callable, Iterator, List, Optional, Sequence, Tuple
+from typing import Any, Callable, Iterator, List, Optional, Sequence, Tuple
 
 import numpy as np
 import torch
 import torch.nn.functional as F
+from sklearn.base import BaseEstimator
+from sklearn.neighbors import KNeighborsClassifier as _KNeighborsClassifier
```
Comment (ronuchit) [inline src/ml_models.py:17]: better naming for these might be `SKLearnKNeighborsClassifier` and `SKLearnKNeighborsRegressor`

File: src/ml_models.py (hunk 1 above)
Comment (tomsilver) [reply src/ml_models.py:729]: because not all classes need to normalize, so they might only want to implement `fit` and `predict`

**Review by amburger66 — APPROVED** [review]
